pkg/il/builder.go

```diff
@@ -0,0 +1,281 @@
+// Copyright 2017 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package il
+
+import (
+	"fmt"
+)
+
+// Builder is used for building function bodies in a programmatic way.
+type Builder struct {
+	strings *StringsTable
+	body    []uint32
+	labels  map[string]uint32
+	fixups  map[string][]uint32
+}
+
+// NewBuilder creates a new Builder instance. It uses the given StringsTable when it needs to
+// allocate ids for strings.
+func NewBuilder(s *StringsTable) *Builder {
+	return &Builder{
+		strings: s,
+		body:    make([]uint32, 0),
```

Comment:
To skip the silly allocation pattern here for the common case, how about initializing the array to 32 entries so that it typically never gets reallocated?